File: backend/src/main/java/com/example/entity/DeliveryList.java
package com.example.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDate;

@Entity
@Table(name = "delivery_list")
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString(exclude = {"order"})
public class DeliveryList {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "delivery_id")
    private Integer deliveryId;

    @Column(name = "date_arrived")
    private LocalDate dateArrived;

    @Column(name = "payment_method")
    private String paymentMethod;

    @OneToOne
    @JoinColumn(name = "order_id")
    private Order order;
}

File: frontend/src/app/model/class/DeliveryList.ts


export class DeliveryList {
  id: string;
  paymentMethod: string;
  getDate: string;
  dateArrived: string;


  constructor(id: string, paymentMethod: string, getDate: string, dateArrived: string) {
    this.id = id;
    this.paymentMethod = paymentMethod;
    this.getDate = getDate;
    this.dateArrived = dateArrived;
  }
}
